.github/workflows/newman.yml

```diff
@@ -0,0 +1,170 @@
+on:
+  workflow_call:
+    inputs:
+      collection_path:
+        required: true
+        type: string
+      container_name:
+        required: true
+        type: string
+      centreon_url:
+        required: true
+        type: string
+      centreon_image:
+        required: true
+        type: string
+      dependencies_lock_file:
+        required: true
+        type: string
+    secrets:
+      registry_username:
+        required: true
+      registry_password:
+        required: true
+
+jobs:
+  newman-test-list:
+    runs-on: ubuntu-22.04
+
+    outputs:
+      collections: ${{ steps.set_collections.outputs.collections }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: List Postman Collections and Environments
+        id: set_collections
+        run: |
+          collection_path="${{ inputs.collection_path }}"
+          collections=$(find "$collection_path" -type f -name "*.postman_collection.json" -printf "%P\n" | sort | jq -R -s -c 'split("\n")[:-1]')
+          echo "collections=$collections" >> $GITHUB_OUTPUT
+          echo "GITHUB_OUTPUT contents:"
+          cat $GITHUB_OUTPUT
+
+  newman-test-run:
+   needs: [newman-test-list]
+   runs-on: ubuntu-22.04
+   continue-on-error: true
+
+   strategy:
+    fail-fast: false
+    matrix:
+      feature: ${{ fromJson(needs.newman-test-list.outputs.collections) }}
+
+   name: ${{ matrix.feature }}
+
+   defaults:
+    run:
+      shell: bash
+      working-directory: centreon/tests/rest_api
+   env:
+    CONTAINER_NAME: ${{ inputs.container_name }}
+    CENTREON_URL: ${{ inputs.centreon_url }}
+    CENTREON_IMAGE: ${{ inputs.centreon_image }}
+
+   steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+
+    - uses: pnpm/action-setup@v2
+      with:
+        version: 7
+        run_install: false
+
+    - uses: actions/setup-node@v3
+      with:
+          node-version: 18
+          cache: pnpm
+          cache-dependency-path: ${{ inputs.dependencies_lock_file }}
+
+    - name: Install Dependencies for tests/rest_api
+      run: pnpm install --frozen-lockfile
+      shell: bash
+      env:
+        CYPRESS_INSTALL_BINARY: “0”
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: “1"
+        PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: “true”
+
+    - name: Login to registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}
+        username: ${{ secrets.registry_username }}
+        password: ${{ secrets.registry_password}}
+
+    - name: Start Centreon Web Container
+      run: docker run --name $CONTAINER_NAME -d -p 4000:80 --health-cmd="curl -f $CENTREON_URL/ || exit 1" --health-interval=5s $CENTREON_IMAGE

```

Comment:
get docker image from cache instead of puling it from registry
example behat / cypress workflows